CMakeLists.txt

```diff
@@ -13,7 +13,7 @@ option(DEBUG "Compile with debugging information." OFF)
 option(PROFILE "Compile with profiling information." OFF)
 option(ARMA_EXTRA_DEBUG "Compile with extra Armadillo debugging symbols." OFF)
 option(TEST_VERBOSE "Run test cases with verbose output." OFF)
-option(BUILD_TESTS "Build tests." ON)
+option(BUILD_TESTS "Build tests." OFF)
```

Comment:
I recommend adding an explicit caveat/note here to state that building tests will take a long time.  For example:  
`option(BUILD_TESTS "Build tests (NOTE: time consuming)." OFF`

This is to allow people to make an informed choice, and not complain that "mlpack is slow to install".
There are also time-constrained packagers out there which by default think that building tests is always good.

Reply:
Yes, that's a good point!  Done in eb37cbb2a5. :+1: